connmgr.go

```diff
@@ -376,7 +376,6 @@ func (cm *BasicConnMgr) getConnsToClose() []network.Conn {
 		// lock this to protect from concurrent modifications from connect/disconnect events
 		s := cm.segments.get(inf.id)
 		s.Lock()
-
 		if len(inf.conns) == 0 && inf.temp {
 			// handle temporary entries for early tags -- this entry has gone past the grace period
 			// and still holds no connections, so prune it.
```

Comment:
Why don't we need to `delete` from the map in the `else` below? (GitHub doesn't allow me to comment that line).